### ISSUE TITLE
Cannot be installed on OS X?

### ISSUE THREAD

**Issue by sivabudh** [phatina/python-lmiwbem#24]
Hello, 

I'm running OS X El Capitan, and I tried to follow the installation instruction, but when I ran `../configure` I the build tool reported that my platform is not supported.

Screenshot: https://www.evernote.com/l/Af0vrur2ip1L6qmok5klid-pD8_ocxTautY

Am I doing something wrong or is there a plan to support OS X?

**Comment by phatina** [comment]
Hi,

> I'm running OS X El Capitan, and I tried to follow the installation instruction, but when I ran ../configure I the build tool reported that my platform is not supported.

I haven't tried compiling it on OS X, because I have no experience with TOG-Pegasus at this platform. Please, can you verify, if Pegasus is compilable on OS X and report it back?

Once you are able to get TOG-Pegasus to OS X (we link python-lmiwbem with their client library), we can move on and add this platform to supported.

Thanks.

**Comment by phatina** [comment]
From [pegasus/doc/porting.txt](http://cvs.opengroup.org/cgi-bin/viewcvs.cgi/pegasus/doc/porting.txt?rev=1.6&content-type=text/vnd.viewcvs-markup):

```
Notes for Mac OS X (10.2.X or Higher) :

   gcc version 3.3.

   use gnutar for extracting pegasus source on Mac OS X.

   Used dynamic libraries for g++.

   Add -dynamiclib -fPIC to compiler line.

   to be able to build Pegasus, dlcompat needed to be installed on Mac OS X.
   dlcomapt is available  from :

   http://www.opendarwin.org/projects/dlcompat/
```

It seems, it should be doable. I can't help much here; I don't own any OS X machine...

**Comment by deleisha** [comment]
@phatina, 
Should be doable, We have not done pegasus on Mac for a very very long time.

**Comment by rchateauneu** [comment]
Hi

By the way I never managed to build WMIMapper , due to severe compile
errors. It comes with OpenPegasus which builds fine on Windows apart a
couple of small problems, easy to fix.

And also, is there information about registering providers on Windows ?

Thanks

**Comment by deleisha** [comment]
@rchateauneu 
WMIMapper, IIRC, is meant for Windows ONLY. It does the translation from CIM to WMI.
So we can safely ignore WMIMapper for Mac.

**Comment by rchateauneu** [comment]
Yes, this is right, WMIMapper is made for Windows only of course. But
still it does not build on it. And also, Hewlett-Packard distributes a
free-ware (But closed source) version of WMIMapper, which therefore
cannot be the same code.

Any explanation please ?

**Comment by deleisha** [comment]
IBM use to have WMIMapper on top of pegasus WMIMapper, Not sure about HP if their is totally independent one or uses pegasus mapper internally. One time we requested for open sourcing those code but unfortunately that did not happen.

**Comment by rchateauneu** [comment]
"One time we requested for open sourcing"

Indeed, legally they have to give it.

Something which amazed me also is this patent, clearly related to WMI Mapper:

"Translation of technology-agnostic management commands into multiple
management protocols"

http://www.freepatentsonline.com/8914482.html

**Comment by deleisha** [comment]
Hmm, Good to see Mike Brasher in the filling. He is one of the first architect of pegasus and including WMIMapper
